miniscope_io/stream_daq.py

```diff
@@ -235,11 +246,14 @@ def _uart_recv(
         log_uart_buffer = BitArray([x for x in uart_bites])
 
         try:
-            while 1:
+            while not self.terminate.is_set():
                 # read UART data until preamble and put into queue
                 uart_bites = serial_port.read_until(pre_bytes)
                 log_uart_buffer = [x for x in uart_bites]
-                serial_buffer_queue.put(log_uart_buffer)
+                try:
+                    serial_buffer_queue.put(log_uart_buffer, block=True, timeout=queue_put_timeout)
+                except queue.Full:
+                    self.logger.warning("Serial buffer queue full, skipping buffer.")
```

Comment:
Yup, that sounds good. Just FYI, I put this here for debug use, and this sacrifice isn't really happening at the moment, though it's possible, as you say. I suspected some buffer might be dropping, so I put this here to check in the log (this should raise an error anyway, but just in case).